core/arch/arm/kernel/tee_i2c.c

```diff
@@ -0,0 +1,58 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright 2020 Foundries Ltd <jorge@foundries.io>
+ *
+ * Brief   Access to linux i2c bus
+ *
+ */
+#include <kernel/tee_i2c.h>
+#include <kernel/thread.h>
+#include <mm/mobj.h>
+#include <optee_rpc_cmd.h>
+#include <string.h>
+
+#define I2C_BUFFER_LENGTH 512
+static struct mobj *mobj;
+
+TEE_Result tee_i2c_transfer(struct TEE_I2CRequest *req, size_t *bytes)
+{
+	TEE_Result res = TEE_SUCCESS;
+
+	if (!bytes || !req)
+		return TEE_ERROR_BAD_PARAMETERS;
+
+	if (req->bufferLen > I2C_BUFFER_LENGTH) {
+		EMSG("increase buffer size (%d)", (uint32_t)req->bufferLen);
```

Comment:
Normally you'd print "buffer to large" or something like that.

Reply:
ok